packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -212,7 +212,7 @@ export class AutoScalingGroup extends cdk.Construct implements cdk.ITaggable, el
 
     launchConfig.addDependency(this.role);
 
-    const minSize = props.minSize || 1;
+    const minSize = props.minSize !== undefined ? props.minSize : 1;
     const maxSize = props.maxSize || 1;
     const desiredCapacity = props.desiredCapacity || 1;
```

Comment:
Same change in other lines?

Reply:
Sure -- realistically thought wasn't sure if `maxSize` or `desiredCapacity` would ever be 0 though, but happy to include the same change.